src/main/java/org/valdroz/vscript/Variant.java

```diff
@@ -325,6 +326,33 @@ public String toString() {
         }
     }
 
+    /**
+     * Rounds the given value to the specified number of decimal places, following standard rounding rules.
+     * Handles nulls, non-numeric, negatives, and very large numbers gracefully.
+     *
+     * @param value Variant or numeric value to round (expressions supported)
+     * @param decimalPlaces number of decimal places to round to
+     * @return Variant containing rounded value (int if decimalPlaces==0, else float)
+     */
+    public static Variant round(Variant value, int decimalPlaces) {
+
+        if (decimalPlaces < 0) {
+            throw new EvaluationException("Invalid decimal places, must be a positive number: " + decimalPlaces);
+        }
+
+      if (value == null) {
+          value = (decimalPlaces == 0)
+                  ? Variant.fromInt(0)
+                  : Variant.fromBigDecimal(BigDecimal.ZERO);
+      }
+
+        BigDecimal num = value.asNumeric();
+        if (num == null) {
+            num = Variant.nullVariant().asNumeric();
```

Comment:
`num = Variant.nullVariant().asNumeric();` must be replaced with `return Variant.nullVariant();`